src/source-updater.js

```diff
@@ -219,55 +350,149 @@ export default class SourceUpdater extends videojs.EventTarget {
+      sourceUpdater: this,
+      action: actions.addSourceBuffer(type, codec),
+      name: 'addSourceBuffer'
+    });
+  }
+
+  /**
+   * call abort on a source buffer.
+   *
+   * @param {string} type
+   *        The type of source buffer to call abort on.
+   */
+  abort(type) {
+    pushQueue({
+      type,
+      sourceUpdater: this,
+      action: actions.abort(type),
+      name: 'abort'
+    });
+  }
+
+  /**
+   * Call removeSourceBuffer and remove a specific type
+   * of source buffer on the mediaSource.
+   *
+   * @param {string} type
+   *        The type of source buffer to remove.
+   */
+  removeSourceBuffer(type) {
+    if (!this.canRemoveSourceBuffer()) {
+      videojs.log.error('removeSourceBuffer is not supported!');
       return;
     }
 
-    if (codecs.audio) {
-      const mime = getMimeForCodec(codecs.audio);
+    pushQueue({
+      type: 'mediaSource',
+      sourceUpdater: this,
+      action: actions.removeSourceBuffer(type),
+      name: 'removeSourceBuffer'
+    });
+  }
 
-      this.audioBuffer = this.mediaSource.addSourceBuffer(mime);
-      this.audioBuffer.removing = false;
-      this.logger_(`created SourceBuffer ${mime}`);
-    }
+  /**
+   * Whether or not the removeSourceBuffer function is supported
+   * on the mediaSource.
+   *
+   * @return {boolean}
+   *          if removeSourceBuffer can be called.
+   */
+  canRemoveSourceBuffer() {
+    return window.MediaSource &&
+      window.MediaSource.prototype &&
+      typeof window.MediaSource.prototype.removeSourceBuffer === 'function';
+  }
 
-    if (codecs.video) {
-      const mime = getMimeForCodec(codecs.video);
+  /**
+   * Whether or not the changeType function is supported
+   * on our SourceBuffers.
+   *
+   * @return {boolean}
+   *         if changeType can be called.
+   */
+  canChangeType() {
+    return window.SourceBuffer &&
+        window.SourceBuffer.prototype &&
+        typeof window.SourceBuffer.prototype.changeType === 'function';
+  }
 
-      this.videoBuffer = this.mediaSource.addSourceBuffer(mime);
-      this.videoBuffer.removing = false;
-      this.logger_(`created SourceBuffer ${mime}`);
+  /**
+   * Call the changeType function on a source buffer, given the code and type.
+   *
+   * @param {string} type
+   *        The type of source buffer to call changeType on.
+   *
+   * @param {string} codec
+   *        The codec string to change type with on the source buffer.
+   */
+  changeType(type, codec) {
+    if (!this.canChangeType()) {
+      videojs.log.error('changeType is not supported!');
+      return;
     }
 
-    this.trigger('ready');
-    this.start_();
+    pushQueue({
+      type,
+      sourceUpdater: this,
+      action: actions.changeType(codec),
+      name: 'changeType'
+    });
   }
 
-  start_() {
-    this.started_ = true;
+  /**
+   * Whether or not the codecSwitch function is supported.
+   *
+   * @return {boolean}
+   *         if codecSwitch can be called
+   */
+  canCodecSwitch() {
```

Comment:
Originally I had tried to support ie 11 type switching and audio/video to audio only switching by re-creating the mediaSource and re-adding source buffer but ultimately abandoned it since it was going to be too much work for this effort. I think that we can likely remove this function now and re-add it at another time when something like that comes up.